### PR TITLE
Tag resource with remove changes

### PR DESCRIPTION
**Pull request by ChanochShayner**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Changes in this PR - 
This PR came to solve the issue that we are not identifying the change when we are just removing code from the resources,
By adding a check of  - `previous-commit-lines` `>` `current-commit-lines`.

fixes: https://github.com/bridgecrewio/yor/issues/377

### CODE REVIEW

File: src/common/gitservice/blame.go [hunk 1]
```diff
@@ -19,16 +21,50 @@ type GitBlame struct {
...
 	for line := startLine; line <= endLine; line++ {
 		if line >= len(blameResult.Lines) {
-			logger.Warning(fmt.Sprintf("Index out of bound on parsed file %s", filePath))
+			logger.Warning(fmt.Sprintf("Index out of bound on parsed file %s", relativeFilePath))
 			return &gitBlame
 		}
 		gitBlame.BlamesByLine[line+1] = blameResult.Lines[line]
+
+		// Check if the line has been removed in the current state of the file
+		if previousBlameResult != nil && len(previousBlameResult.Lines) > len(blameResult.Lines) {
```
Comment (gruebel) [inline src/common/gitservice/blame.go:62]: not sure about this check `len(previousBlameResult.Lines) > len(blameResult.Lines)` will this always be true or are there cases where is `<=` but still the specific line was removed?

**Review by rotemavni — COMMENTED** [review]
@ChanochShayner would be great to see a test / add to the description of this PR what this change does 🙏

File: src/common/gitservice/blame.go [hunk 2]
```diff
@@ -19,16 +21,50 @@ type GitBlame struct {
...
+		return nil, nil
+	}
+	parentIter := commit.Parents()
+	previousCommit, err := parentIter.Next()
+	if err != nil {
+		return nil, nil
+	}
+
+	var previousBlameResult *git.BlameResult
+	result, _ := g.PreviousBlameByFile.Load(filePath)
```
Comment (omryMen) [inline src/common/gitservice/blame.go:43]: ignoring the error is not recommended

File: src/common/gitservice/blame.go [hunk 3]
```diff
@@ -19,16 +21,50 @@ type GitBlame struct {
...
+	previousBlameResult = result.(*git.BlameResult)
+	return previousBlameResult, previousCommit
+}
+
+func NewGitBlame(relativeFilePath string, filePath string, lines structure.Lines, blameResult *git.BlameResult, g *GitService) *GitBlame {
```
Comment (omryMen) [inline src/common/gitservice/blame.go:48]: name g is not informative => gitSvc

**Review by rotemavni — APPROVED** [review]

**Pull request merged**